plugin-server/src/worker/ingestion/person-state.ts

```diff
@@ -712,7 +712,7 @@ export class DeferredPersonOverrideWriter {
      * @param lockId the lock identifier/key used to ensure that only one
      *               process is updating the overrides at a time
      */
-    constructor(private postgres: PostgresRouter, private lockId: number) {}
+    constructor(private postgres: PostgresRouter, private lockId: number = 567) {}
```

Comment:
Yes, this is intentional; this is for the advisory lock acquired here that ensures only one process at a time is processing pending overrides to avoid race conditions in transitive overrides processing from overlapping transactions (also described on line 712 above):

https://github.com/PostHog/posthog/blob/7e2065a43cb218768cf522e3d9b659397d52ac37/plugin-server/src/worker/ingestion/person-state.ts#L758-L768

It's annoying that these are numbers and some varchar/text type that could be more descriptive, but that's what the Postgres API provides: https://www.postgresql.org/docs/current/explicit-locking.html#ADVISORY-LOCKS

The value could also be provided via a setting, but I couldn't think of a reason to make that configurable so I avoided that for now. I guess this could be a class property rather than an instance property though, I had initially made it an instance property so that the functional test runner could be running while other unit/integration tests were run — but thinking about it more, those are still sharing the same tables on the test database so they probably _should_ conflict with each other. 

It's possible that this could get replaced with a table lock on the overrides (not pending overrides) table but I want to defer that at least until the immediate overrides path is fully removed so that we don't lock the overrides table by accident while it's still being used in the ingest consumer transaction during the changeover.

Reply:
Made this static with e2a7487ccd652453cc9288a6ae883f7c4ac7c870.

Reply:
It's minor, but it may be worth putting a "yep, this is how the PG API works, we're just picking a consistent int yadda yadda" line alongside your existing comment. I have to admit I read it and was still head-tilt at the magic number.